sql/src/main/java/io/crate/expression/scalar/string/StringLeftRightFunction.java

```diff
@@ -0,0 +1,130 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.scalar.string;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.crate.data.Input;
+import io.crate.expression.scalar.ScalarFunctionModule;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Scalar;
+import io.crate.metadata.TransactionContext;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+
+import java.util.List;
+import java.util.Locale;
+
+public class StringLeftRightFunction extends Scalar<String, Object> {
+
+    private static final String LEFT_NAME = "left";
+    private static final String RIGHT_NAME = "right";
+
+    public static void register(ScalarFunctionModule module) {
+        module.register(LEFT_NAME, new Resolver(LEFT_NAME));
+        module.register(RIGHT_NAME, new Resolver(RIGHT_NAME));
+    }
+
+    private final FunctionInfo info;
+
+    private StringLeftRightFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public String evaluate(TransactionContext txnCtx, Input[] args) {
+        assert args.length == 2 : String.format(Locale.ENGLISH,
+                                                "number of arguments must be 2, got %d instead",
+                                                args.length);
+        String str = (String) args[0].value();
+        Number len = (Number) args[1].value();
+        if (str == null || len == null) {
+            return null;
+        }
+
+        switch (info.ident().name()) {
+            case LEFT_NAME:
+                return left(str, len.intValue());
+            case RIGHT_NAME:
+                return right(str, len.intValue());
+            default:
+                throw new UnsupportedOperationException(String.format(Locale.ENGLISH,
+                                                                      "not yet implemented: %s",
+                                                                      info.ident().name()));
+        }
+    }
+
+    @VisibleForTesting
+    static String left(String str, int len) {
+        if (str == null || str.isEmpty() || len == 0) {
```

Comment:
`str == null` cannot be true here as this was already checked upfront by https://github.com/crate/crate/pull/8879/files#diff-7a2324ee3f0c0f2c921e7ffeb1740058R70 (it also correctly results in `null` there).
We could also move this empty string or 0 length up under that check as it would be same for both functions.

Reply:
Correct, will update.